test/BitStringAddresses.jl

```diff
@@ -460,3 +469,104 @@ end
+        @test_throws ArgumentError OccupationNumberFS{3}(1.1, 2, 3)
+        @test_throws ArgumentError OccupationNumberFS{3}(-1, 2, 3)
+        @test_throws ArgumentError OccupationNumberFS{3}(1, 2, 300)
+    end
+
+    @testset "Test OccupationNumberFS with BoseFS input" begin
+        fs = BoseFS(1, 2)
+        @test isa(OccupationNumberFS(fs), OccupationNumberFS{2, UInt8})
+        fs = BoseFS(1, 333)
+        @test isa(OccupationNumberFS(fs), OccupationNumberFS{2,UInt16})
+    end
+
+    @testset "Test printing and parsing OccupationNumberFS" begin
+        fs = OccupationNumberFS(1, 2, 3, 0, 1, 20, 3, 2, 5, 0, 1)
+        @test eval(Meta.parse(repr(fs))) == fs
+        @test parse_address(sprint(show, fs; context=:compact => true)) == fs
+
+        for T in [UInt8, UInt16, UInt32, UInt64, UInt128]
+            fs = OccupationNumberFS{11,T}(1, 2, 3, 0, 1, 20, 3, 2, 5, 0, 1)
+            @test eval(Meta.parse(repr(fs))) == fs
+            @test parse_address(sprint(show, fs; context=:compact => true)) == fs
+        end
+
+        @test_throws ArgumentError parse_address("fs\"|1 2 3⟩{-8}\"")
+        @test_throws ArgumentError parse_address("fs\"|1 2 3⟩{129}\"")
+    end
+
+    ofs = OccupationNumberFS{3,UInt8}(1, 2, 3)
+    @testset "Test destroy function" begin
+        ofs_after_destroy, val_before_destroy = destroy(ofs, 2)
+        @test ofs_after_destroy.onr == SVector{3,UInt8}(1, 1, 3)
+        @test val_before_destroy == 2
+    end
+
+    @testset "Test create function" begin
+        ofs_after_create, val_after_create = create(ofs, 2)
+        @test ofs_after_create.onr == SVector{3,UInt8}(1, 3, 3)
+        @test val_after_create == 3
+    end
+
+    @testset "Test excitation function" begin
+        c = (1,)
+        d = (2,)
+        fs_after_excitation, sqrt_accu = excitation(ofs, c, d)
+        @test fs_after_excitation.onr == SVector{3,UInt8}(2, 1, 3)
+        @test sqrt_accu ≈ √4
+
+        # indexing with BoseFSIndex
+        i, j = find_mode(ofs, (1, 2))
+        @test j ==  BoseFSIndex(occnum=2, mode=2, offset=2)
+        @test excitation(ofs, (i,), (j,)) == (fs_after_excitation, sqrt_accu)
+    end
+
+    @testset "Test properties of OccupationNumberFS" begin
+        @test num_modes(ofs) == 3
+        @test num_particles(ofs) == 6
+        @test num_occupied_modes(ofs) == 3
+        @test onr(ofs) == ofs.onr == SVector{3,UInt8}(1, 2, 3)
+        @test occupation_number_representation(ofs) == onr(ofs)
+        @test onr(reverse(ofs)) == reverse(onr(ofs))
+        lfs = OccupationNumberFS{6}([1 0 0; 1 1 0])
+        @test onr(lfs, LadderBoundaries(2, 3)) == [1 0 0; 1 1 0]
+        @test num_occupied_modes(lfs) == length(occupied_modes(lfs)) == 3
+        @test OccupiedModeMap(lfs) == collect(occupied_modes(lfs))
+        b1, b2 = BoseFS(1, 6), BoseFS(3, 4)
+        o1, o2 = OccupationNumberFS(b1), OccupationNumberFS(b2)
+        @test (o1 < o2) == (b1 < b2)
+    end
+
```

Comment:
```suggestion
@testset "OccupationNumberFS functions" begin
    @testset "OccupationNumberFS with SVector input" begin
        @test OccupationNumberFS(SVector{3, UInt8}(1, 2, 3)) isa OccupationNumberFS{3, UInt8}
        @test_throws ArgumentError OccupationNumberFS(SVector(-1, 2, 3))
        @test_throws ArgumentError OccupationNumberFS(SVector(1, 2, 300))
        @test OccupationNumberFS(SVector{3,UInt16}(1, 2, 300)) isa OccupationNumberFS{3,UInt16}
    end

    @testset "OccupationNumberFS with multiple arguments" begin
        @test isa(OccupationNumberFS(1, 2, 3), OccupationNumberFS{3, UInt8})
        @test_throws ArgumentError OccupationNumberFS(1.1, 2, 3)
        @test_throws ArgumentError OccupationNumberFS(-1, 2, 3)
        @test_throws ArgumentError OccupationNumberFS(1, 2, 300)
    end

    @testset "OccupationNumberFS with M and multiple arguments" begin
        @test OccupationNumberFS{3}([1, 2, 3]) == OccupationNumberFS{3,UInt8}(1, 2, 3)
        @test_throws ArgumentError OccupationNumberFS{3}(1.1, 2, 3)
        @test_throws ArgumentError OccupationNumberFS{3}(-1, 2, 3)
        @test_throws ArgumentError OccupationNumberFS{3}(1, 2, 300)
    end

    @testset "OccupationNumberFS with BoseFS input" begin
        fs = BoseFS(1, 2)
        @test isa(OccupationNumberFS(fs), OccupationNumberFS{2, UInt8})
        fs = BoseFS(1, 333)
        @test isa(OccupationNumberFS(fs), OccupationNumberFS{2,UInt16})
    end

    @testset "Printing and parsing OccupationNumberFS" begin
        fs = OccupationNumberFS(1, 2, 3, 0, 1, 20, 3, 2, 5, 0, 1)
        @test eval(Meta.parse(repr(fs))) == fs
        @test parse_address(sprint(show, fs; context=:compact => true)) == fs

        for T in [UInt8, UInt16, UInt32, UInt64, UInt128]
            fs = OccupationNumberFS{11,T}(1, 2, 3, 0, 1, 20, 3, 2, 5, 0, 1)
            @test eval(Meta.parse(repr(fs))) == fs
            @test parse_address(sprint(show, fs; context=:compact => true)) == fs
        end

        @test_throws ArgumentError parse_address("fs\"|1 2 3⟩{-8}\"")
        @test_throws ArgumentError parse_address("fs\"|1 2 3⟩{129}\"")
    end

    ofs = OccupationNumberFS{3,UInt8}(1, 2, 3)
    @testset "Destroy function" begin
        ofs_after_destroy, val_before_destroy = destroy(ofs, 2)
        @test ofs_after_destroy.onr == SVector{3,UInt8}(1, 1, 3)
        @test val_before_destroy == 2
    end

    @testset "Create function" begin
        ofs_after_create, val_after_create = create(ofs, 2)
        @test ofs_after_create.onr == SVector{3,UInt8}(1, 3, 3)
        @test val_after_create == 3
    end

    @testset "Excitation function" begin
        c = (1,)
        d = (2,)
        fs_after_excitation, sqrt_accu = excitation(ofs, c, d)
        @test fs_after_excitation.onr == SVector{3,UInt8}(2, 1, 3)
        @test sqrt_accu ≈ √4

        # indexing with BoseFSIndex
        i, j = find_mode(ofs, (1, 2))
        @test j ==  BoseFSIndex(occnum=2, mode=2, offset=2)
        @test excitation(ofs, (i,), (j,)) == (fs_after_excitation, sqrt_accu)
    end

    @testset "Properties of OccupationNumberFS" begin
        @test num_modes(ofs) == 3
        @test num_particles(ofs) == 6
        @test num_occupied_modes(ofs) == 3
        @test onr(ofs) == ofs.onr == SVector{3,UInt8}(1, 2, 3)
        @test occupation_number_representation(ofs) == onr(ofs)
        @test onr(reverse(ofs)) == reverse(onr(ofs))
        lfs = OccupationNumberFS{6}([1 0 0; 1 1 0])
        @test onr(lfs, LadderBoundaries(2, 3)) == [1 0 0; 1 1 0]
        @test num_occupied_modes(lfs) == length(occupied_modes(lfs)) == 3
        @test OccupiedModeMap(lfs) == collect(occupied_modes(lfs))
        b1, b2 = BoseFS(1, 6), BoseFS(3, 4)
        o1, o2 = OccupationNumberFS(b1), OccupationNumberFS(b2)
        @test (o1 < o2) == (b1 < b2)
    end

```
I'd prefer it like this, but we can keep it as is if you want. I don't know why the diff is not displayed nicely, but I just removed the word "Test" from all testset names.

Reply:
Haha, that was co-pilot's verbosity. Most of the test were auto-generated.